File: system-test/java/src/test/java/com/optivem/shop/systemtest/latest/acceptance/PublishCouponPositiveTest.java
package com.optivem.shop.systemtest.latest.acceptance;

import com.optivem.shop.systemtest.latest.acceptance.base.BaseAcceptanceTest;
import com.optivem.testing.Channel;
import com.optivem.shop.dsl.channel.ChannelType;

import org.junit.jupiter.api.TestTemplate;

class PublishCouponPositiveTest extends BaseAcceptanceTest {

    @TestTemplate
    @Channel(ChannelType.API)
    void shouldPublishCouponSuccessfully() {
        scenario
                .when().publishCoupon()
                    .withCode("SAVE10")
                    .withDiscountRate(0.10)
                .then().shouldSucceed();
    }
}

File: system-test/dotnet/SystemTests/Latest/AcceptanceTests/PublishCouponPositiveTest.cs
using SystemTests.Latest.AcceptanceTests.Base;
using Optivem.Testing;

namespace SystemTests.Latest.AcceptanceTests;

public class PublishCouponPositiveTest : BaseAcceptanceTest
{
    [Theory]
    [ChannelData(ChannelType.API)]
    public async Task ShouldPublishCouponSuccessfully(Channel channel)
    {
        await Scenario(channel)
            .When().PublishCoupon().WithCode("SAVE10").WithDiscountRate(0.1m)
            .Then().ShouldSucceed();
    }
}
